docs/pain_points.md

```diff
@@ -37,13 +37,23 @@ Crashes the pipeline when the destination table does not exist.
 
 ## `FileIO.Write`
 
-Acknowledges messages in PubSub before they are written due to a required
-`GroupByKey` transform. This also effects `BigQueryIO.Write` in batch mode.
+Acknowledges messages in PubSub before they are written to accumulate data
+across multiple [bundles] and produce reasonably sized files. Possible
+workaround being investigated in [#380]. This also effects `BigQueryIO.Write`
+in batch mode.
+
+[bundles]: https://beam.apache.org/documentation/execution-model/#bundling-and-persistence
+[#380]: https://github.com/mozilla/gcp-ingestion/issues/380
 
 ## `PubsubIO.Write`
 
 Does not support dynamic destinations.
 
+Does not support [`NestedValueProvider`] for destinations, which is needed to
```

Comment:
AFAIK, this limitation only exists on Dataflow in streaming mode, where Google moves the implementation into the shuffler. Probably worth being explicit on that.